### PR TITLE
chore(PouchDB-find): loop updates

### PR DESCRIPTION
**Pull request by SourceR85**
Improve readability and performance (slightly) of loops in pouchdb-find.

- refactor iterator loops to for ... of: when `i` was only for element lookup.
- flatten: use native `Array.prototype.flat` is 2x faster than polyfill.
- `uniq` utility -> Array.from(Set): slight perf boost (`Set` is guaranteed to be unique).

Isolated from #8860

### CODE REVIEW

File: packages/node_modules/pouchdb-find/src/utils.js [hunk 1]
```diff
@@ -196,13 +200,7 @@ function arrayEquals(arr1, arr2) {
 }
 
 function uniq(arr) {
-  const obj = {};
-  for (let i = 0; i < arr.length; i++) {
-    obj['$' + arr[i]] = true;
-  }
-  return Object.keys(obj).map(function (key) {
-    return key.substring(1);
-  });
```
Comment (garethbowen) [inline packages/node_modules/pouchdb-find/src/utils.js:205]: Wow, that's insane...

**Review by garethbowen — APPROVED** [review]

File: packages/node_modules/pouchdb-find/src/adapters/local/find/query-planner.js [hunk 2]
```diff
@@ -32,8 +32,7 @@ const SHORT_CIRCUIT_QUERY = {
 
 function checkFieldInIndex(index, field) {
   const indexFields = index.def.fields.map(getKey);
-  for (let i = 0, len = indexFields.length; i < len; i++) {
-    const indexField = indexFields[i];
+  for (const indexField of indexFields) {
```
[the remark this answers is not in the document]
Comment (SourceR85) [reply packages/node_modules/pouchdb-find/src/adapters/local/find/query-planner.js]: Thanks for the hint 👍️
Just as I was reading this, I had an idea for another shortening:
https://github.com/pouchdb/pouchdb/commit/323e43eabb10bec05151734879f3738ef86a729a

File: packages/node_modules/pouchdb-find/src/utils.js (hunk 1 above)
Comment (SourceR85) [reply packages/node_modules/pouchdb-find/src/utils.js:205]: Pardon?

Comment (garethbowen) [reply packages/node_modules/pouchdb-find/src/utils.js:205]: The old code, not the new stuff!  :)

**Review by garethbowen — APPROVED** [review]
Thanks!

**Review by garethbowen — APPROVED** [review]
Thanks!

**Pull request merged**